src/main/java/com/ncc/aif/ValidateAIF.java

```diff
@@ -63,75 +75,304 @@ private static void loadModel(Model model, CharSource ontologySource) {
+                "-f FILE ...\tvalidate the specified file(s) with a .ttl suffix\n" +
+                "-d DIRNAME\tValidate all .ttl files in the specified directory\n" +
+                "\n" +
+                "Either a file (-f) or a directory (-d) must be specified (but not both).\n" +
+                "Exactly one of --ldc, --program, or --ont must be specified.\n" +
+                "Ontology files can be found in src/main/resources/com/ncc/aif/ontologies:\n" +
+                "- LDC: SeedlingOntology\n" +
+                "- Program: EntityOntology, EventOntology, RelationOntology\n" +
+                "\n" +
+                "For more information, see the AIF README.");
+    }
+
+    // Process an option that takes N files as an argument and return N.
+    private static int processFiles(String[] args, int i, Set<String> fileList) {
+        i++; // skip the switch (-f or -d)
+        boolean done = false;
+        int numArgs = 0;
+        while (i < args.length && !done) {
+            if (args[i].startsWith("-")) {
+                done = true;
+            }
+            else {
+                fileList.add(args[i++]);
+                numArgs++;
+            }
+        }
+        return numArgs; // Can't use fileList.size() just in case use specified same file twice
+    }
+
+    // Process command-line arguments, returning whether or not there were any errors.
+    private static boolean processArgs(String[] args, Set<String>flags, Set<String>domainOntologies,
+                                       Set<String> validationFiles, Set<String> validationDirs) {
+        boolean dashD = false;
+        boolean dashF = false;
+        for (int i = 0; i < args.length; i++) {
+            final String arg = args[i];
+            final String strippedArg = arg.trim();
+            switch (strippedArg) {
+                case "-h":
+                case "--help" :
+                    return false;
+                case "--ldc" :
+                    flags.add("LDC");
+                    break;
+                case "--program" :
+                    flags.add("PROGRAM");
+                    break;
+                case "--both" :
+                    flags.add("LDC");
+                    flags.add("PROGRAM");
+                    break;
+                case "--nist" :
+                    flags.add("NIST");
+                    break;
+                case "--ont" :
+                    int numFiles = processFiles(args, i, domainOntologies);
+                    if (numFiles == 0) {
+                        System.err.println("ERROR: --ont requires at least one ontology file to be specified.");
+                        return false;
+                    }
+                    i += numFiles;
+                    break;
+                case "-f" :
+                    if (dashD) {
+                        System.err.println("ERROR: Please specify either -d or -f, but not both.");
```

Comment:
It certainly could.  It could also easily support -d allowing a list of directories (you probably saw I added comments in the code how we'd support it).  For that matter, it could just take a bunch of paths and if it's a .ttl file, validate it, and if it's a directory, validate all of the .ttl files in it.  That is, combine -f and -d.
Anyway, these are all reasonable changes, but I'd suggest we see what kinds of comments we get at the AIDA Collab level when Craig circulates something about the validator.  There may well be another pass at this as we fold in user requests.

Reply:
Oh, also -f and -d are designed to match the anticipated use cases:  -d for TA1 output (single AIF file per document) and -f for TA2/TA3 output (single very large AIF file).  But we can still combine them.